src/main/java/org/folio/service/impl/StagingUserService.java

```diff
@@ -65,41 +67,71 @@ public Future<User> mergeOrCreateUserFromStagingUser(String stagingUserId, Strin
-          }
-        }).compose(userUpdatesStorage-> Future.succeededFuture(userUpdatesStorage.getNewEntity()))
+      .withTrans(conn -> fetchStagingUserById(conn, stagingUserId)
+        .compose(stagingUser -> {
+
+          List<Future<?>> requiredFetches = List.of(
+            fetchHomeAddressType(conn),
+            fetchRemotePatronGroup(conn, userId, stagingUser.getMinor()),
+            fetchBasicMinorPatronGroup(conn, stagingUser.getMinor())
+          );
+
+          return Future.all(requiredFetches).compose(compositeFuture -> {
+            AddressType homeAddress = compositeFuture.resultAt(0);
+            String homeAddressId = homeAddress.getId();
+            Usergroup remotePatronGroup = compositeFuture.resultAt(1);
+            Usergroup basicMinorPatronGroup = compositeFuture.resultAt(2);
+
+            return getCreateOrUpdateUser(userId, conn, stagingUser, homeAddressId, basicMinorPatronGroup,
+              remotePatronGroup).compose(userUpdatesStorage -> deleteStagingUser(conn, stagingUserId,
+              userUpdatesStorage));
+          });
+        })).onSuccess(userUpdatesStorage -> publishUserEvent(userId, userUpdatesStorage))
+      .compose(userUpdatesStorage -> succeededFuture(userUpdatesStorage.getNewEntity()))
       .onFailure(t -> log.error("mergeOrCreateUserFromStagingUser:: Merge or creation failed for stagingUserId {}, userId {}: {}",
         stagingUserId, userId, t.getMessage()));
   }
 
+  private Future<StagingUserUpdatesStorage<User>> getCreateOrUpdateUser(String userId, Conn conn,
+    StagingUser stagingUser, String homeAddressId, Usergroup basicMinorPatronGroup, Usergroup remotePatronGroup) {
+    return (userId != null ?
+      updateExistingUserDetailsFromStagingUser(stagingUser, userId, conn, homeAddressId, basicMinorPatronGroup)
+      : createNewUserFromStagingUser(stagingUser, conn, homeAddressId, remotePatronGroup, basicMinorPatronGroup));
+  }
+
+  private void publishUserEvent(String userId, StagingUserUpdatesStorage<User> userUpdatesStorage) {
+    if (userId == null) {
+      userEventPublisher(vertxContext, okapiHeaders)
+        .publishCreated(userUpdatesStorage.getNewEntity().getId(), userUpdatesStorage.getNewEntity());
+    } else {
+      userEventPublisher(vertxContext, okapiHeaders)
+        .publishUpdated(userId, userUpdatesStorage.getOldEntity(), userUpdatesStorage.getNewEntity());
+    }
+  }
+
   private Future<AddressType> fetchHomeAddressType(Conn conn) {
     log.debug("fetchHomeAddressType:: fetching address of type home");
     return fetchEntityByCriterion(conn, ADDRESS_TYPE, HOME, AddressType.class, ADDRESS_TYPE_TABLE,
       HOME_ADDRESS_TYPE_NOT_FOUND);
   }
 
-  private Future<Usergroup> fetchRemotePatronGroupOnlyIfUserIdNotPresent(Conn conn, String userId) {
+  private Future<Usergroup> fetchRemotePatronGroup(Conn conn, String userId, Boolean minor) {
```

Comment:
please also update the below logging in a next line:
```suggestion
  log.debug("fetchRemotePatronGroup:: fetching Remote patron group with userId {}", userId);
```

Reply:
fixed=>https://github.com/folio-org/mod-users/tree/6cdf07958f3a0368dbf015627d2a49ce03334095